clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/models/output/SCV.scala

```diff
@@ -153,4 +160,81 @@ object SCV {
       case comment       => Array(normalizeComment(comment))
     }
   }
+
+  /**
+    * Helper method to extract the accession id from a raw SCV Msg.
+    * Throws an exception if none is found, as that is an illegal state.
+    */
+  def extractAccessionId(rawScv: Msg): String = {
+    rawScv
+      .extract("ClinVarAccession", "@Accession")
+      .getOrElse {
+        throw new IllegalStateException(s"Found an SCV with no Accession ID: $rawScv")
+      }
+      .str
+  }
+
+  /**
+    * Helper method to extract the numeric id from a raw SCV Msg.
+    * Throws an exception if none is found, as that is an illegal state.
+    */
+  def extractNumericId(rawScv: Msg): String = {
+    rawScv
+      .extract("@ID")
+      .getOrElse {
+        throw new IllegalStateException(s"Found an SCV with no numeric ID: $rawScv")
+      }
+      .str
+  }
+
+  /**
+    * Helper method to find the ID of the corresponding VCV Trait Set for the given SCV,
+    * which is identified by the scvAccessionId.
+    *
+    * Given an SCV ID, we first find the corresponding SCVTraitSet. We then find the SCVTraits within
+    * that SCVTraitSet, and find the clinvar traitIds that they point to. Using these clinvar traitIds,
+    * we then find the VCVTraitSet that contains the VCVTraits that have the same clinvar traitIds. We
+    * then return the ID of the found VCVTraitSet.
+    */
+  def findRelatedVcvTraitSetId(
+    scvTraitSets: ArrayBuffer[WithContent[SCVTraitSet]],
+    scvAccessionId: String,
+    scvTraits: ArrayBuffer[WithContent[SCVTrait]],
+    vcvTraitSets: ArrayBuffer[WithContent[VCVTraitSet]]
+  ): Option[String] = {
+    scvTraitSets
+    // filter scvTraitSets down to the ones for the current scv
+      .find(_.data.id == scvAccessionId)
+      .flatMap { traitSet =>
+        // need to use traitSet to find the right scv traits first
+        val scvTraitIds = scvTraits.filter { `trait` =>
+          traitSet.data.scvTraitIds.contains(`trait`.data.id)
+        }.flatMap(_.data.traitId)
```

Comment:
We should be able to massage our control-flow so we can entirely skip this "which SCV traits / trait-sets are relevant" search. I'll comment in `VariationArchive`